ui/app/AppLayouts/Profile/helpers/ProfileShowcaseDirtyState.qml

```diff
@@ -0,0 +1,83 @@
+import QtQml 2.15
+
+import StatusQ 0.1
+import StatusQ.Core.Utils 0.1
+
+import SortFilterProxyModel 0.2
+
+/**
+  * Building block for managing temporary state in the "Profile Showcase"
+  * functionality. Provides combining raw source model (like e.g. communities
+  * model or accounts model) with lean showcase model (providing info regarding
+  * visibility and position), managing dirty state (visibility, position) and
+  * providing two output models - one containing visible items sorted by
+  * position, second one containing hidden items.
+  */
+QObject {
+    property alias sourceModel: joined.leftModel
+    property alias showcaseModel: joined.rightModel
+
+    /**
+      * Model holding elements from 'sourceModel' intended to be visible in the
+      * showcase, sorted by 'position' role. Includes roles from both input models.
+      */
+    readonly property alias visibleModel: visible
+
+    /**
+      * Model holding elements from 'sourceModel' intended to be hidden, no
+      * sorting applied. Includes roles from both input models.
+      */
+    readonly property alias hiddenModel: hidden
+
+    function currentState() {
+        return writable.currentState()
+    }
+
+    function setVisibility(key, visibility) {
+        writable.setVisibility(key, visibility)
+    }
+
+    function changePosition(key, to) {
+        writable.changePosition(key, to)
+    }
```

Comment:
I'm wondering if these operations (`setVisibility` and/or `changePosition`) could also be done in the delegate as well. I think the SFPM will pass the setData calls to the writable model.
It would look something like:
Delegate {
    onEvent: model.visibility = newVisibility
}

It would simplify things even more.

Reply:
I think that this should be done outside of the delegate. Because the logic may easily change and it may be necessary to do some other actions along with setting visibility. Even now, when unhiding there is also additional logic of setting position to last.

Reply:
Agree, I'm just worried the js model logic could backfire at us in terms of performance. It's just something we can consider if this happens. Probably the `model` variable could be shared by the UI component and let this component decide how to act on it.

But NVM, let's give this a spin and see how it will behave with the nim models. Could be an unnecessary optimisation.